Sorting out lapack & blas interfaces

wenfei-li opened the pull request.

Commit:
move Cblacs.h to module_base and rename to cblacs_connector.h

Commit:
add some comments in lapack_connector.h; remove DGER from myfunc.h

Commit:
removed unused wrapper ILAENV, zgemv, and ZHPEV
from myfunc.h

Commit:
removed unused wrapper dsyev
put zhegv and zhegvx together
in lapack_connector.h

caic99 reviewed (approved):
lgtm, @dyzheng please review.

Commit:
the ultimate goal is to have unified interfaces
to matrix & array operations, in this commit:
1. removed unused 'ddots'
2. removed unused subroutines from myfunc.h
3. minor modification to the two header files (blas & lapack)

wenfei-li commented:
> lgtm, @dyzheng please review.

I've added some more changes :P

Commit:
add some comments in header file
and moved some subroutines around

Commit:
turn on deepks in dynamic.yml
